bindings/profilers/cpu.cc

```diff
@@ -0,0 +1,268 @@
+#include <queue>
+
+#include <v8-profiler.h>
+#include <v8.h>
+#include <node.h>
+#include <uv.h>
+
+#include "../per-isolate-data.hh"
+#include "../location.hh"
+#include "cpu.hh"
+
+namespace dd {
+
+CpuProfiler::CpuProfiler()
+  : isolate_(v8::Isolate::GetCurrent()),
+    async(new uv_async_t()),
+    code_map(CodeMap::For(isolate_)),
+    samples(Nan::New<v8::Array>()),
+    sampler_running(false) {
+  // TODO: Move symbolizer worker to a separate class?
+  uv_async_init(Nan::GetCurrentEventLoop(), async, Run);
+  uv_unref(reinterpret_cast<uv_handle_t*>(async));
```

Comment:
Could you comment around this for libuv noobs ? Do we want to decouple lifetime of node from the profiler ?